.github/workflows/build.yml

```diff
@@ -0,0 +1,154 @@
+name: "Build"
+
+on:
+  pull_request:
+    paths-ignore:
+      - doc/**
+  push:
+    paths-ignore:
+      - doc/**
+    branches:
+      - master
+      - v*.*
+
+jobs:
+  phpstan:
+    name: PHPStan
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-versions: [ '7.4' ]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup PHP with pecl extension
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+
+      - name: Get composer cache directory
+        id: composercache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composercache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('composer.json') }}
+          restore-keys: ${{ runner.os }}-composer-
+
+      - name: Install dependencies
+        run: composer install --prefer-dist
```

Comment:
I am not sure if it cannot cause issues, combined with composer cache and no lock file. Shouldn't it be `composer update`?

Reply:
It should be the same in the case composer.lock is present. caching shouldn't affect it.